ember_debug/view-debug.js

```diff
@@ -4,54 +4,56 @@ import RenderTree from 'ember-debug/libs/render-tree';
 import ViewInspection from 'ember-debug/libs/view-inspection';
 import bound from 'ember-debug/utils/bound-method';
 
-import { readOnly } from 'ember-debug/utils/ember/object/computed';
-
-export default DebugPort.extend({
-  namespace: null,
-
-  adapter: readOnly('namespace.adapter'),
-  objectInspector: readOnly('namespace.objectInspector'),
-
-  portNamespace: 'view',
-
-  messages: {
-    getTree({ immediate }) {
-      this.sendTree(immediate);
-    },
-
-    showInspection({ id, pin }) {
-      this.viewInspection.show(id, pin);
-    },
-
-    hideInspection() {
-      this.viewInspection.hide();
-    },
-
-    inspectViews({ inspect }) {
-      if (inspect) {
-        this.startInspecting();
-      } else {
-        this.stopInspecting();
-      }
-    },
-
-    scrollIntoView({ id }) {
-      this.renderTree.scrollIntoView(id);
-    },
-
-    inspectElement({ id }) {
-      this.renderTree.inspectElement(id);
-    },
-
-    contextMenu() {
-      let { lastRightClicked } = this;
-      this.lastRightClicked = null;
-      this.inspectNearest(lastRightClicked);
-    },
-  },
+export default class extends DebugPort {
+  get adapter() {
+    return this.namespace?.adapter;
+  }
+  get objectInspector() {
+    return this.namespace?.objectInspector;
+  }
+
+  static {
```

Comment:
Do we need static and prototype stuff here or could this just go on the class instance?

Reply:
this is needed on prototype.
as `setupOrRemovePortListeners` accesses the `.messages` and `.portNamespace` in the constructor.
Which will happen before the sub class can set them up in their own constructor.
We could change this by either passing the messages and portNamespace as constrcutor param.
or by calling `setupOrRemovePortListeners` in the sub classes instead

Reply:
actually, I can make it work the same way as Ember Core Object.
```
const obj = new this();
obj.init();
return obj;
````